bigquery/google/cloud/bigquery/table.py

```diff
@@ -1334,8 +1334,21 @@ def _to_dataframe_tabledata_list(self, dtypes):
         """Use (slower, but free) tabledata.list to construct a DataFrame."""
         column_names = [field.name for field in self.schema]
         frames = []
+
+        # report progress if tqdm installed
+        try:
+            from tqdm import tqdm
+            pbar = tqdm(desc="Downloading", total=self.total_rows, unit="rows")
```

Comment:
Does this do something sensible when `total_rows` is not populated? As discussed in https://github.com/googleapis/google-cloud-python/pull/7217 `total_rows` is `None` until iteration starts.

Reply:
Good catch, I'll update `pbar.total` in the loop if its unset. `tqdm` handles this just fine.